docs/asset-store/docs/assets/asset-metadata-service-openapi.yaml

```diff
@@ -0,0 +1,134 @@
+openapi: 3.0.0
+info:
+  description: The purpose of this application is to provide an API to extract metadata from Front Matter YAML block in text files.
+  title: Asset Metadata Service API
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0
+  version: 0.0.1
+paths:
+  /extract:
+    post:
+      description: Extracts metadata from text files sent in `multipart/form-data` form. As a field name, specify relative or absolute path to the file.
+      tags:
+        - extract
+      summary: Extracts metadata from files.
+      operationId: extractReq
+      requestBody:
+        content:
+          multipart/form-data:
+            schema:
+              type: object
+              additionalProperties:
```

Comment:
what are those `additionalProperties` for? are they supported by the service somehow?

Reply:
It's the only way to specify that fields in the form are dynamic, where key is string, and value is a file.

Reply:
after reading https://swagger.io/docs/specification/describing-request-body/file-upload/
this looks better in my opinion
```
      requestBody:
        content:
          multipart/form-data:
            schema:
              type: object
              properties:
                filename:
                  type: array
                  items:
                    type: string
                    format: binary
```
but I also do not plan to argue here ;)

Reply:
Yes, I wanted to do so. But unfortunately in file metadata I have only filename with extension. What if someone sends array of files from paths `~/testdata/test.yaml` and `~/foo/test.yaml`? I cannot differentiate them, and the result would contain two array items with the same fileName.

Reply:
this is sad 😞